src/com/facebook/buck/android/MergeAndroidResourcesStep.java

```diff
@@ -346,26 +351,77 @@ void writePerPackageRDotJava(
       }
 
       String packageName = entry.getValue();
-      for (String line : linesInSymbolsFile) {
-        Optional<RDotTxtEntry> parsedEntry = RDotTxtEntry.parse(line);
-        Preconditions.checkState(parsedEntry.isPresent(), "Should be able to match '%s'.", line);
-
-        // We're only doing the remapping so Roboelectric is happy and it is already ignoring the
-        // id references found in the styleable section.  So let's do that as well so we don't have
-        // to get fancier than is needed.  That is, just re-enumerate all app-level resource ids
-        // and ignore everything else, allowing the styleable references to be messed up.
-        RDotTxtEntry resource = parsedEntry.get();
+
+      int index = 0;
+      while (index < linesInSymbolsFile.size()) {
+        RDotTxtEntry resource = getResourceAtIndex(linesInSymbolsFile, index);
+        index++;
```

Comment:
your reuse of the index as the offset of the next resource later in the code is a bit confusing. Use a regular for loop and assign index +1 to an appropriately named variable.